src/index.ts

```diff
@@ -18,20 +18,24 @@ export default class TypedStorage<T> implements ITypedStorage<T> {
 
   constructor(options: TypedStorageOptions = { storage: 'localStorage' }) {
     this.storage = typeof window !== 'undefined' ? window[options.storage] : global[options.storage];
+
+    if (!this.storage) {
+      throw Error('Web Storage API not found.');
```

Comment:
If you throw an error won't the entire app crash? If someone uses this library for nothing critical then the app will crash even though it would work without storage. Or am I wrong?

Reply:
If the error is not handled it could crash a app. But I think this would be the way to go, so the developer can decide what to do if no storage is available. This is also closer to using the storage API directly, if it is not present `window.localStorage.getItem('someKey')` would also throw an error (`Uncaught TypeError: window.localStorage is undefined`).

An alternative would be `console.error` or a flag to switch between throwing an error or just logging it.